File: Week-02/readability/readability.c
#import <stdio.h>
#import <cs50.h>
#import <string.h>
#import <ctype.h>
#import <math.h>

int main(void)
{
    // Ask user for text
    string text = get_string("Text: ");
    int str = strlen(text);

    // Store letters, words, sentences
    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int i = 0; i < str; i++)
    {
        // If text is alphabetical and isn't '\0' count as letter
        if (isalpha(text[i]) && text[i] != '\0')
        {
            letters++;
        }
        // If text is space count as word
        if (text[i] == ' ')
        {
            words++;
        }
        // If text is .?! count as sentence
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }

    // printf("Total letters: %i\n", letters);
    // printf("Total words: %i\n", words);
    // printf("Total sentences: %i\n", sentences);

    float L = ((float)letters / (float)words) * 100;
    float S = ((float)sentences / (float)words) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade_index = round(index);

    if (grade_index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade_index);
    }
}

File: Week-06/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Ask user for text
    text = get_string("Text: ")

    # Store letter, words, sentences
    letters = 0
    words = 1
    sentences = 0

    # Loop through text to find letters, word, sentences
    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1
        elif text[i] == " ":
            words += 1
        elif text[i] == "." or text[i] == "?" or text[i] == "!":
            sentences += 1

    # Coleman-Liau formula
    L = float(letters) / float(words) * 100
    S = float(sentences) / float(words) * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    grade_index = round(index)

    # Grade text input
    if grade_index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade_index}")


if __name__ == "__main__":
    main()
